Problem: upgrade to tonic 0.6 fails

```
   Compiling tonic v0.6.0
391
error[E0277]: the trait bound `hyper::common::exec::Exec: hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not satisfied
392
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-0.6.0/src/transport/server/mod.rs:510:18
393
    |
394
510 |                 .with_graceful_shutdown(signal)
395
    |                  ^^^^^^^^^^^^^^^^^^^^^^ the trait `hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not implemented for `hyper::common::exec::Exec`
396
    |
397
    = help: the following implementations were found:
398
              <hyper::common::exec::Exec as hyper::common::exec::NewSvcExec<I, N, S, E, W>>
399

400
error[E0277]: `(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)` cannot be shared between threads safely
401
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-0.6.0/src/transport/server/mod.rs:510:18
402
    |
403
510 |                 .with_graceful_shutdown(signal)
404
    |                  ^^^^^^^^^^^^^^^^^^^^^^ `(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)` cannot be shared between threads safely
405
    |
406
    = help: the trait `Sync` is not implemented for `(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)`
407
    = note: required because of the requirements on the impl of `Sync` for `Unique<(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)>`
408
    = note: required because it appears within the type `Box<(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)>`
409
    = note: required because it appears within the type `Pin<Box<(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)>>`
410
    = note: required because it appears within the type `UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>`
411

412
error[E0277]: the trait bound `hyper::common::exec::Exec: hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not satisfied
413
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-0.6.0/src/transport/server/mod.rs:508:13
414
    |
415
508 | /             server
416
509 | |                 .serve(svc)
417
510 | |                 .with_graceful_shutdown(signal)
418
511 | |                 .await
419
    | |______________________^ the trait `hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not implemented for `hyper::common::exec::Exec`
420
    |
421
    = help: the following implementations were found:
422
              <hyper::common::exec::Exec as hyper::common::exec::NewSvcExec<I, N, S, E, W>>
423
    = note: required because of the requirements on the impl of `futures_core::Future` for `hyper::server::shutdown::Graceful<impl Accept, MakeSvc<<L as Layer<S>>::Service, IO>, F, hyper::common::exec::Exec>`
424
note: required by `poll`
425

426
error[E0277]: `(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)` cannot be shared between threads safely
427
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-0.6.0/src/transport/server/mod.rs:508:13
428
    |
429
508 | /             server
430
509 | |                 .serve(svc)
431
510 | |                 .with_graceful_shutdown(signal)
432
511 | |                 .await
433
    | |______________________^ `(dyn HttpBody<Error = Box<(dyn std::error::Error + Send + Sync + 'static)>, Data = bytes::Bytes> + Send + 'static)` cannot be shared between threads safely
```

https://github.com/crypto-org-chain/chainlib-rs/runs/4001208602?check_suite_focus=true

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,7 @@
 name = "chainlib"
 version = "0.1.1"
 authors = ["linfeng yuan <linfeng@crypto.com>"]
-edition = "2018"
+edition = "2021"
 autoexamples = false
 license = "Apache-2.0"
 repository = "https://github.com/crypto-org-chain/chainlib-rslib-rs"
@@ -13,39 +13,35 @@ keywords = ["blockchain", "chain", "cosmos", "tendermint", "transaction"]
 exclude = ["cosmos-sdk-go/**", "tendermint/**"]
 
 [dependencies]
-serde = { version = "1.0", features = ["derive"], optional = true }
-sorted-json = { version = "0.1", optional = true }
-serde_json = { version = "1.0", optional = true }
-tiny-bip39 = "0.8"
-anyhow = { version = "1.0", features = ["std"] }
-hdwallet = "0.3"
-bitcoin_hashes = "0.10"
-rand = "0.8"
-base64 = "0.13"
+serde = { version = "1.0.130", features = ["derive"], optional = true }
+sorted-json = { version = "0.1.0", optional = true }
+serde_json = { version = "1.0.68", optional = true }
+tiny-bip39 = "0.8.2"
+anyhow = { version = "1.0.44", features = ["std"] }
+hdwallet = "0.3.0"
+bitcoin_hashes = "0.10.0"
+base64 = "0.13.0"
 stdtx = "0.5.0"
-thiserror = "1.0"
-tokio = { version = "1.12", features = ["full"] }
-ledger-transport = { version = "0.8", optional = true }
-ledger-transport-hid = { version = "0.8", optional = true }
-ledger-zondax-generic = { version = "0.8", optional = true }
-zx-bip44 = "0.1"
+thiserror = "1.0.30"
+tokio = { version = "1.12.0", features = ["full"] }
+ledger-transport = { version = "0.8.0", optional = true }
+ledger-transport-hid = { version = "0.8.0", optional = true }
+ledger-zondax-generic = { version = "0.8.0", optional = true }
+zx-bip44 = { version = "0.1.0", optional = true }
 reqwest = { version = "0.11.6", features = ["json"] }
-async-trait = "0.1"
-log = "0.4"
-tonic = { version = "0.5", optional = true }
-eyre = "0.6.4"
-signature = "1.3.2"
-prost = { version = "0.8", optional = true }
-prost-types = { version = "0.8", optional = true }
-hex = { version = "0.4", optional = true }
+async-trait = "0.1.51"
+log = "0.4.14"
+tonic = { version = "0.6.0", optional = true }
+eyre = "0.6.5"
+prost = { version = "0.9.0", optional = true }
+prost-types = { version = "0.9.0", optional = true }
+hex = { version = "0.4.3", optional = true }
 
 [build-dependencies]
-prost = "0.8"
-prost-build = "0.9"
-tonic = "0.5"
-tonic-build = "0.5"
-regex = "1"
-walkdir = "2"
+prost-build = "0.9.0"
+tonic-build = "0.6.0"
+regex = "1.5.4"
+walkdir = "2.3.2"
 
 [[example]]
 name = "amino"
@@ -67,6 +63,7 @@ ledger = [
     "ledger-transport-hid",
     "ledger-transport",
     "hex",
+    "zx-bip44",
 ]
 amino = ["serde", "serde_json", "sorted-json"]
 grpc = ["tonic", "prost", "prost-types"]

